nx/lib/nx/type.ex

```diff
@@ -195,6 +195,31 @@ defmodule Nx.Type do
   def to_complex({:f, 64}), do: {:c, 128}
   def to_complex(_type), do: {:c, 64}
 
+  @doc """
+  Converts the given type to a float with the minimum size
+  necessary.
+
+  ## Examples
+
+      iex> Nx.Type.to_float({:s, 8})
+      {:f, 32}
+      iex> Nx.Type.to_float({:s, 64})
+      {:f, 32}
+      iex> Nx.Type.to_float({:bf, 16})
+      {:f, 32}
+      iex> Nx.Type.to_float({:c, 64})
+      {:f, 32}
+      iex> Nx.Type.to_float({:c, 128})
+      {:f, 64}
+      iex> Nx.Type.to_float({:f, 32})
+      {:f, 32}
+      iex> Nx.Type.to_float({:f, 64})
+      {:f, 64}
+  """
+  def to_float({:f, size}), do: {:f, size}
+  def to_float({:c, s}), do: {:f, div(s, 2)}
+  def to_float(_type), do: {:f, 32}
```

Comment:
My suggestion is to call this function `to_real`. Because it is all about going from complex to something in the real domain.

Reply:
> This is going to convert bf16 to f32, and it should not.
> My suggestion is to call this function to_real. Because it is all about going from complex to something in the real domain.

My inital plan was to convert this to float regardless of the previous type, but since what we are trying to do is achieve an acceptable representation for real numbers, I think we can rename this to `to_real` like José suggested.